socorro/unittest/lib/test_vertools.py

```diff
@@ -2,29 +2,24 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
-import socorro.lib.ver_tools as vtl
+import pytest
 
-phs = vtl._padding_high_string
-pl = vtl._padding_list
+from socorro.lib import ver_tools
 
-tests = [('3',            [3, phs, 0, phs] + pl * 3, '3'),
-         ('3.',           [3, phs, 0, phs] + pl * 3, '3'),
-         ('3.0',          [3, phs, 0, phs] + pl * 3, '3'),
-         ('3.0.0',        [3, phs, 0, phs] + pl * 3, '3'),
-         ('3.5',          [3, phs, 0, phs,
-                           5, phs, 0, phs] + pl * 2, '3.5'),
-         ('3.5pre',       [3, phs, 0, phs,
-                           5, 'pre', 0, phs] + pl * 2, '3.5pre'),
-         ('3.5b3',        [3, phs, 0, phs,
-                           5, 'b', 3, phs] + pl * 2, '3.5b3'),
 
-         ('3.6.4plugin3', [3, phs, 0, phs,
-                           6, phs, 0, phs,
-                           4, 'plugin', 3, phs] + pl, '3.6.4plugin3'),
-         ]
+phs = ver_tools._padding_high_string
+pl = ver_tools._padding_list
 
 
-def test_normalize():
-    for ver, expected, ver2 in tests:
-        got = vtl.normalize(ver)
-        assert got == expected, "expected %s, but got %s" % (expected, got)
+@pytest.mark.parametrize('ver, expected, ver2', [
+    ('3', [3, phs, 0, phs] + pl * 3, '3'),
+    ('3.', [3, phs, 0, phs] + pl * 3, '3'),
+    ('3.0', [3, phs, 0, phs] + pl * 3, '3'),
+    ('3.0.0', [3, phs, 0, phs] + pl * 3, '3'),
+    ('3.5', [3, phs, 0, phs, 5, phs, 0, phs] + pl * 2, '3.5'),
+    ('3.5pre', [3, phs, 0, phs, 5, 'pre', 0, phs] + pl * 2, '3.5pre'),
+    ('3.5b3', [3, phs, 0, phs, 5, 'b', 3, phs] + pl * 2, '3.5b3'),
+    ('3.6.4plugin3', [3, phs, 0, phs, 6, phs, 0, phs, 4, 'plugin', 3, phs] + pl, '3.6.4plugin3'),
+])
+def test_normalize(ver, expected, ver2):
```

Comment:
concise. I dig it.

Reply:
I don't really understand this test and certainly couldn't debug it without fleshing out some of the values. Now that I'm looking at it, `ver2` isn't used at all.

I tossed around auditing the tests more thoroughly. Are they readable? Do they test what we think they're testing? But that's a lot of work and I'm not sure we get much value out of it right now.